proxy.go

```diff
@@ -58,29 +57,20 @@ func (hpd *httpProxyDialer) Dial(network string, addr string) (net.Conn, error)
 	}
 
 	if err := connectReq.Write(conn); err != nil {
-		if err := conn.Close(); err != nil {
-			log.Printf("httpProxyDialer: failed to close connection: %v", err)
-		}
-		return nil, err
+		return nil, errors.Join(conn.Close(), err)
 	}
 
 	// Read response. It's OK to use and discard buffered reader here becaue
 	// the remote server does not speak until spoken to.
 	br := bufio.NewReader(conn)
 	resp, err := http.ReadResponse(br, connectReq)
 	if err != nil {
-		if err := conn.Close(); err != nil {
-			log.Printf("httpProxyDialer: failed to close connection: %v", err)
-		}
-		return nil, err
+		return nil, errors.Join(conn.Close(), err)
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		if err := conn.Close(); err != nil {
-			log.Printf("httpProxyDialer: failed to close connection: %v", err)
-		}
 		f := strings.SplitN(resp.Status, " ", 2)
-		return nil, errors.New(f[1])
+		return nil, errors.Join(conn.Close(), err, errors.New(f[1]))
```

Comment:
`err` does not belong here.   Primary error first.

    return nil, errors.Join(errors.New(f[1]), conn.Close())